Remove jnp.DeviceArray since it is deprecated.

mariecwhite opened the pull request:
The latest version of JAX deprecates jax.numpy.DeviceArray. When using this version with Transformers, we get this error when instantiating FlaxBertModel: `module 'jax.numpy' has no attribute 'DeviceArray'`

mariecwhite committed:
Remove jnp.DeviceArray since it is deprecated.

sgugger commented:
cc @sanchit-gandhi

sanchit-gandhi commented on src/transformers/models/bert/modeling_flax_bert.py:
```diff
@@ -1677,7 +1677,7 @@ def __call__(
     module_class = FlaxBertForCausalLMModule
 
-    def prepare_inputs_for_generation(self, input_ids, max_length, attention_mask: Optional[jnp.DeviceArray] = None):
+    def prepare_inputs_for_generation(self, input_ids, max_length, attention_mask = None):
```
Would be great to keep the type hint here but update it as required
```suggestion
    def prepare_inputs_for_generation(self, input_ids, max_length, attention_mask: Optional[jax.Array] = None):
```

mariecwhite replied:
Thanks Sanchit, I've replaced all instances of `jax.DeviceArray` with `jax.Array`.

sanchit-gandhi reviewed (commented):
Hey @mariecwhite, thanks for flagging this deprecation! The JAX [change logs](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-7-march-27-2023) suggest using `jax.Array` as a substitute for `jnp.DeviceArray`. Could we maintain a type hint, but update it to `jax.Array` here? 

I did a quick search and found 38 instances of `jnp.DeviceArray` in the library, including this one in Flax BERT. Would you like to perform a library wide Flax update and fix them all in one go? Thanks!

mariecwhite committed:
Replace all instances of jnp.DeviceArray with jax.Array

ayaka14732 commented on src/transformers/models/bart/modeling_flax_bart.py:
```diff
@@ -1467,8 +1467,8 @@ def prepare_inputs_for_generation(
         max_length,
-        attention_mask: Optional[jnp.DeviceArray] = None,
-        decoder_attention_mask: Optional[jnp.DeviceArray] = None,
+        attention_mask: Optional[jnp.Array] = None,
+        decoder_attention_mask: Optional[jnp.Array] = None,
```
```suggestion
        attention_mask: Optional[jax.Array] = None,
        decoder_attention_mask: Optional[jax.Array] = None,
```

Should be `jax.Array` instead of `jnp.Array`.

mariecwhite replied:
Oh yes, thanks. Changed.

ayaka14732 commented:
@sanchit-gandhi This pull request should be prioritised since the affected Flax models are currently not usable at all.

amyeroberts commented:
@mariecwhite Thanks for opening this PR! Running `make style` and pushing the changes will resolve the code quality CI checks

sanchit-gandhi committed:
Update src/transformers/models/bert/modeling_flax_bert.py

sanchit-gandhi commented:
Hey @mariecwhite - it seems there is an issue with your CircleCI permissions, meaning the tests won't run!
Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)? Let me know if you encounter any issues!

mariecwhite commented:
> Hey @mariecwhite - it seems there is an issue with your CircleCI permissions, meaning the tests won't run! Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)? Let me know if you encounter any issues!

I just updated my CircleCI permissions and rebased.

mariecwhite commented:
CircleCI is still failing for me. Since this is a priority, I'm happy to let https://github.com/huggingface/transformers/pull/25275 get merged instead of this.

sgugger commented:
Forced circleCI to run by also pushing the same branch on the main fork of Transformers. Should solve the tests issues (at least if there are no further commits needed 😅 )

sgugger reviewed (approved):
@sanchit-gandhi You haven't approved yet, so I'll let you merge if you are happy with the changes.

sanchit-gandhi reviewed (approved):
Thanks @mariecwhite! Good for me to merge!
